vllm/v1/core/sched/scheduler.py

```diff
@@ -1018,8 +1050,10 @@ def finish_requests(
                 waiting_requests_to_remove.append(request)
 
         # Remove all requests from queues at once for better efficiency
-        for request in running_requests_to_remove:
-            self.running.remove(request)
+        if running_requests_to_remove:
+            # Convert list to set for efficient removal
+            self.running = remove_all(self.running,
+                                      set(running_requests_to_remove))
```

Comment:
Instead of converting to a set here, could you just change this to be a set where it's declared above